### DISCUSSION TITLE
How to make arguments with underscores instead of dashes?

### DISCUSSION THREAD

**Discussion by AreopagX** [fastapi/typer#1049] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer

def main(local_rank: Optional[int] = 0):
    print(f"local_rank: {local_rank}")
    # do fancy neural network stuff

if __name__ == "__main__":
    typer.run(main)
```

### Description

Hello everyone,

I'm currently using Typer for a project with neural networks. I'd like to start my main script using torch.distributed.launch, which passes the argument "local_rank" to the main script. 

`python main.py --local_rank=0`

However, when doing so Typer complains that this option is unknown. That is due to the standardized  conversion of "_" to "-". How can I make my script accept the CLI option "local_rank"?

There is an related issue #341 , however, this deals with command names. In this case, it is about argument names AND I can't influence the way they are passed to my script.

### Operating System

Linux

### Typer Version

0.12.5

### Python Version

3.10.14

**Comment by YuriiMotov** [comment]
There is no option to configure this for now, but you can use the following "hack" (from [mentioned issue](https://github.com/fastapi/typer/issues/341#issuecomment-1686401587)):

```py
from typing import Optional

import typer

typer.main.get_command_name = lambda name: name  # Override the name conversion function

def main(local_rank: Optional[int] = 0):
    print(f"local_rank: {local_rank}")
    # do fancy neural network stuff

if __name__ == "__main__":
    typer.run(main)
```

Note, it will also change naming convention for commands!
